keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -237,9 +237,9 @@ class SqlArtifactRepository(
         .also { versions ->
           // FIXME: remove special handling for Docker
           return if (artifact is DockerArtifact) {
-            filterDockerVersions(artifact, versions)
+            filterDockerVersions(artifact, versions, limit)
           } else {
-            versions
+            versions.subList(0, Math.min(versions.size, limit))
```

Comment:
Do we expect any significant performance benefit if we don't actually limit the query? I get that it would be difficult to do that given the sort method.

Reply:
I'm not sure how to measure the performance here.. I was struggling with the best way to limit that list, we can do it at the API level as well if you think it's better?

Reply:
I think it's good to have the limit here because it makes subsequent optimization easier. I really hope we can get to a place where we could use a limit on the query itself. In this case we're reading all the entries and hydrating objects, then discarding all but `limit`. I'd expect some kind of improvement just based on the size of the response from the API endpoint, but limiting the query would be optimal.

Reply:
Just piping in here to agree with @robfletcher. We really do need to get in the habit of limiting the size all of our queries at the SQL level or we're going to see performance degradation over time.

The sooner we are able to implement version sorting in the SQL query itself, the better off we're going to be.